src/components/number-input/number-input.vue

```diff
@@ -0,0 +1,247 @@
+  helpertext?: string
+  maxlength?: number
+  maxvalue?: number
+  direction?: 'horizontal' | 'vertical'
+  increment?: number
+};
+
+const emit = defineEmits(['change', 'update:modelValue']);
+
+const props = withDefaults(defineProps<InputProps>(), {
+  validationText: 'validation text',
+  isValid: null,
+  placeholder: undefined,
+  label: '',
+  disabled: false,
+  helpertext: '',
+  modelValue: 0,
+  maxlength: 3,
+  maxvalue: 999,
+  direction: 'horizontal',
+  increment: 1,
+});
+
+const maxLengthRef: Ref = ref(props.maxlength);
+onBeforeMount(() => {
+  if (maxLengthRef.value > 14) maxLengthRef.value = 14;
+});
+
+
+function emitInput(e: HTMLInputElement)  {
+  if (e) {
+ const comparevalue = parseInt(e.value)
+  if ( comparevalue > props.maxvalue) {
+    e.value = props.maxvalue.toString();
+  }
+  if (e.value == '' ) {
+    e.value = '0';
+  }
+  e.value = e.value.slice(0, maxLengthRef.value);
+  const emitValue = parseInt(e.value.slice(0, maxLengthRef.value));
+
+  if (emitValue !== props.modelValue) {
+    emit('update:modelValue', emitValue);
+  }
+}
+}
+
+const verticalAlignment = props.direction == 'vertical' ? 'verticalStyle' : '';
+const labelRef = ref(props.label+ ' '+ uuidv4());
+
+
+function increment() {
+  const updatedValue = props.modelValue + props.increment;
+  if (updatedValue > props.maxvalue) {
+    emit('update:modelValue', props.maxvalue);
+  }
+  if (
+    updatedValue.toString().length <= maxLengthRef.value &&
+    updatedValue <= props.maxvalue
+  )
+    emit('update:modelValue', props.modelValue + props.increment);
+}
+
+function decrement() {
+  const updatedValue = props.modelValue - props.increment;
+  if (updatedValue < 0) {
+    emit('update:modelValue', 0);
+  } else if (props.modelValue > 0)
+    emit('update:modelValue', props.modelValue - props.increment);
+}
+
+const horizontalWidthCalculation = (maxLengthRef.value * 12).toString();
+const verticalWidthCalculation = (maxLengthRef.value * 12).toString();
+const verticalWidthCalculationAlignment = (
+  verticalWidthCalculation + 40
+).toString();
+
+// dynamic component import
+const AsyncIcon = computed(() => {
+  {
+    const Icon = defineAsyncComponent(
+      () => import('@/components/icons/icons.vue')
+    );
+    return Icon;
+  }
+});
+</script>
+
+<template>
+  <div class="numbers-input__div">
+    <div :class="['topWrapper', props.disabled ? 'disabled' : '']">
+      <label :for="labelRef" class="h3">
+        {{ props.label }}
+      </label>
+    </div>
+    <div
+      class="alignment__container"
+      :style="{
+        width:
+          props.direction == 'vertical'
+            ? verticalWidthCalculationAlignment + ' px'
+            : '',
+      }"
+    >
+      <div
+        v-if="props.direction == 'vertical'"
+        :style="{
+          width:
+            props.direction === 'vertical'
+              ? verticalWidthCalculation + 'px'
+              : '',
+        }"
+        :class="[
+          'vertical__incrementDiv',
```

Comment:
There's some inconsistency in the namning format for classnames, for example "numbers-input__div", "topWrapper", "vertical__incrementDiv". Would be nicer if they kept to the same style, I think we've talked about BEM before?